extensions/PhabBugz/lib/Util.pm

```diff
@@ -0,0 +1,310 @@
+            }
+        );
+    }
+
+    return request( 'differential.revision.edit', $data );
+}
+
+sub get_project_phid {
+    my $project = shift;
+
+    my $data = {
+        queryKey    => 'all',
+        constraints => {
+            name => $project
+        }
+    };
+
+    my $result = request( 'project.search', $data );
+    return undef
+      unless ( exists $result->{result}{data} && @{ $result->{result}{data} } );
+
+    return $result->{result}{data}[0]{phid};
+}
+
+sub create_project {
+    my ( $project, $description, $members ) = @_;
+
+    my $data = {
+        transactions => [
+            { type => 'name',        value => $project },
+            { type => 'description', value => $description },
+            { type => 'edit',        value => 'admin' },
+            { type => 'join',        value => 'admin' },
+            { type => 'view',        value => 'admin' },
+            { type => 'icon',        value => 'group' },
+            { type => 'color',       value => 'red' }
+        ]
+    };
+
+    my $result = request( 'project.edit', $data );
+    return $result->{result}{object}{phid};
+}
+
+sub set_project_members {
+    my ( $project_id, $phab_user_ids ) = @_;
+
+    my $data = {
+        objectIdentifier => $project_id,
+        transactions => [ { type => 'members.set', value => $phab_user_ids } ]
+    };
+
+    my $result = request( 'project.edit', $data );
+    return $result->{result}{object}{phid};
+}
+
+sub get_members_by_bmo_id {
+    my $users = shift;
+
+    my $data = { accountids => [ map { $_->id } @$users ] };
+
+    my $result = request( 'bmoexternalaccount.search', $data );
+    return [] if ( !$result->{result} );
+
+    my @phab_ids;
+    foreach my $user ( @{ $result->{result} } ) {
+        push( @phab_ids, $user->{phid} );
+    }
+
+    return \@phab_ids;
+}
+
+sub request {
+    my ( $method, $data ) = @_;
+    my $request_cache = Bugzilla->request_cache;
+    my $params        = Bugzilla->params;
+
+    my $ua = $request_cache->{phabricato_ua};
+    unless ($ua) {
+        $ua = $request_cache->{phabricator_ua} =
+          LWP::UserAgent->new( timeout => 10 );
+        if ( $params->{proxy_url} ) {
+            $ua->proxy( 'https', $params->{proxy_url} );
+        }
+        $ua->default_header(
+            'Content-Type' => 'application/x-www-form-urlencoded' );
+    }
+
+    my $phab_api_key = $params->{phabricator_api_key};
+    ThrowUserError('invalid_phabricator_api_key') unless $phab_api_key;
+    my $phab_base_uri = $params->{phabricator_base_uri};
+    ThrowUserError('invalid_phabricator_uri') unless $phab_base_uri;
+
+    my $full_uri = $phab_base_uri . '/api/' . $method;
+
+    $data->{__conduit__} = { token => $phab_api_key };
+
+    my $response = $ua->post( $full_uri, { params => encode_json($data) } );
+
+    ThrowCodeError( 'phabricator_api_error', { reason => $response->message } )
+      if $response->is_error;
+
+    my $result = decode_json( $response->content );
```

Comment:
This  should be tested when the response isn't valid JSON to make sure it doesn't do something bad.